pythonscript/_godot_io.pxi

```diff
@@ -0,0 +1,163 @@
+            Also tries to get exception information such as, file name, line numer, method name, etc
+            and pass that along to godot_print_error
+        """
+        import traceback
+
+        # we are printing an error message, so we must avoid other errors at all costs,
+        # otherwise the user may never see the error message printed, making debugging a living hell
+        try:
+            if lineno is not None and filename is not None and name is not None:
+                # don't try to get exception info if user provided the details.
+                exc_info = sys.exc_info()
+                tb = exc_info[2]
+                if tb:
+                    tblist = traceback.extract_tb(tb)
+                    if len(tblist) > 0:
+                        lineno = tblist[-1].lineno
+                        filename = tblist[-1].filename
+                        name = tblist[-1].name
+        except:
+            sys.__stderr__.write('Additional errors occured while printing:\n' + traceback.format_exc() + '\n')
+        
+        # default values in case we couldn't get exception info and user have not provided those
+        lineno = lineno or 0
+        filename = filename or '<UNKNOWN>'
+        name = name or '<UNKNOWN>'
+        
+        # TODO: how to handle different encodings?
+        pystr = pystr.encode('utf-8')
+        name = name.encode('utf-8')
+        filename = filename.encode('utf-8')
+        
+        cdef char * c_msg = pystr
+        cdef char * c_name = name
+        cdef char * c_filename = filename
+        cdef int c_lineno = <int>lineno
+        
+        with nogil:
+            gdapi10.godot_print_error(c_msg, c_name, c_filename, c_lineno)
+    
+    @staticmethod
+    def print_override(*objects, sep=' ', end='\n', file=sys.stdout, flush=False):
+        """
+            We need to override the builtin print function to avoid multiple calls to stderr.write.
+            e.g:
+                print(a, b, c, file=sys.stderr)
+                would cause 3 writes to be issued: write(a), write(b) and write(c).
+                Since we are using godot_print_error, that would cause a very weird print to the console,
+                so overriding print and making sure a single call to write is issued solves the problem.
+        """
+
+        if file == GodotIO.get_godot_stderr_io():
+            msg = str(sep).join([str(obj) for obj in objects]) + str(end)
+            GodotIO.godot_print_error_pystr(msg)
+        else:
+            GodotIO._builtin_print(*objects, sep=sep, end=end, file=file, flush=flush)
+    
+    @staticmethod
+    def print_exception_override(etype, value, tb, limit=None, file=None, chain=True):
+        # We override traceback.print_exception to avoid multiple calls to godot_print_error on newlines,
+        # making the traceback look weird
+        from traceback import TracebackException
```

Comment:
Why not move this import on top of the module ?

Reply:
I generally avoid top-level imports for things like this, as to avoid polluting the global scope. (I woudn't want a user to to `from godot import TracebackException` for example). Also, I'm not 100% clear on how cython handles global imports. But ff you think it won't cause a problem I'll move it up.

Reply:
In this case the `_godot` is not designed to be imported by anyone (it is only loaded by libpythonscript when Godot bootstraps godot-python) so no need to worry about 3rd party ;-)

In a more general term, [PEP8 tells us](https://www.python.org/dev/peps/pep-0008/#public-and-internal-interfaces) to use __all__ to document the public API of a module